torchchat/generate.py

```diff
@@ -661,13 +664,22 @@ def chat(
                     False  # Bug with cudagraph trees in this case
                 )
 
+            if self.builder_args.device == "cpu":
+                if generator_args.max_autotune:
+                    kwargs = {"mode": "max-autotune"}
+                    torch._inductor.config.trace.log_autotuning_results = True
+                else:
+                    kwargs = {}
+            else:
+                kwargs = {"mode": "reduce-overhead"}
+
             if self.is_speculative:
                 self.model_forward = torch.compile(
-                    self.model_forward, mode="reduce-overhead", fullgraph=True
+                    self.model_forward, fullgraph=True, **kwargs
                 )
 
             self.decode_one_token = torch.compile(
-                self.decode_one_token, mode="reduce-overhead", fullgraph=True
+                self.decode_one_token, fullgraph=True, **kwargs
             )
 
             if generator_args.compile_prefill:
```

Comment:
Pass `kwargs` to compile_prefill model too?

Reply:
Yes, miss this feature. Create another PR to support. https://github.com/pytorch/torchchat/pull/1112